src/JsonSchema/Constraints/TypeConstraint.php

```diff
@@ -185,6 +185,10 @@ protected function validateType(&$value, $type, $coerce = false)
         }
 
         if ('array' === $type) {
+            if ($coerce) {
+                $value = $this->toArray($value);
```

Comment:
Honestly don't know. We can try without for a while, and see if the rationale for having one ever emerges.